nautobot/core/views/mixins.py

```diff
@@ -249,7 +255,22 @@ def get_permissions_for_model(self, model, actions):
         """
         model_permissions = []
         for action in actions:
-            model_permissions.append(f"{model._meta.app_label}.{action}_{model._meta.model_name}")
+            # Append the base action permission if and only if it is valid and the model provided matches the queryset model.
+            if self.custom_view_base_action:
+                if self.custom_view_base_action not in BASE_ACTIONS:
+                    self.logger.error(
+                        f"Invalid base action '{self.custom_view_base_action}' specified in {self.__class__.__name__}. "
+                    )
+                elif model == self.get_queryset().model:
+                    model_permissions.append(
+                        f"{model._meta.app_label}.{self.custom_view_base_action}_{model._meta.model_name}"
+                    )
```

Comment:
I'm a bit confused why we need this here - `self.get_permissions_for_model` is called by `self.get_required_permission` after calling `self.get_action()`, which *also* has logic for handling `custom_view_base_action`?